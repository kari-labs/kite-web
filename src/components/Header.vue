<template>
    <div>
        <nav-drawer ref="navDrawer"></nav-drawer>
        <v-toolbar app relative dark clipped-left color="primary">
            <v-toolbar-side-icon @click.stop="toggleDrawer" temporary class="secondary-text" aria-label="Toggle side navigation drawer"></v-toolbar-side-icon>
            <v-toolbar-title class="secondary-text" @click="pushDash">NEIT Kite</v-toolbar-title>
            <v-spacer></v-spacer>
            <v-toolbar-items class="hidden-sm-and-down">
                <v-btn flat color="secondary" slot="activator" :to="''" aria-label="Container Dashboard">Container Dashboard</v-btn>
                <v-btn flat color="secondary" aria-label="Report a Bug">Report Bug</v-btn>
            </v-toolbar-items>
        </v-toolbar>
    </div>
</template>

<script>
    export default {
        methods: {
            toggleDrawer() {
                //Toggles the navDrawer open/closed only if the navDrawer component
                //has finished downloading.
                this.$refs.navDrawer.drawer = (typeof this.$refs.navDrawer.drawer == 'undefined') ? false : !this.$refs.navDrawer.drawer
            },
            pushDash() {
                this.$router.push({ name: 'listContainers'})
            }
        },
        components: {
            'nav-drawer': () => import(/* webpackChunkName: "navDrawer" */ '@/components/NavDrawer.vue')
        }
    }
</script>

<style lang="stylus" scoped>
//Fixes bug where Vuetify secondary changes color of background but not text
.v-toolbar__side-icon.secondary-text
    color #FCB741 !important

//Disables ability to select toolbar title text
.v-toolbar__title
    user-select none
</style>
