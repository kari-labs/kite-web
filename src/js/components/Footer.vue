<template>
    <v-footer app absolute inset color="primary" height="auto">
        <v-card flat tile color="primary" class="text-xs-center">
            <v-card-text>
                <v-btn v-for="icon in icons" :key="icon" :aria-label="icon" class="mx-3 secondary-text" icon>
                    <v-icon size="24px" color="secondary">{{icon}}</v-icon>
                </v-btn>
            </v-card-text>
            <v-card-text class="pt-0 secondary-text product-sans">
                Use of this system is limited to authorized persons of New England Institute of Technology.
            </v-card-text>
            <v-divider></v-divider>
            <v-card-text class="secondary-text product-sans">
                &copy;2018 — <strong>Joseph Zurowski</strong> <br />
                Product Sans &copy;2015 — 2018 <strong>google_logo</strong>
            </v-card-text>
        </v-card>
    </v-footer>
</template>

<script>
    export default {
        data: () => ({
            icons: [
                'fab fa-facebook',
                'fab fa-twitter',
                'fab fa-google-plus',
                'fab fa-instagram'
            ]
        })
    }
</script>