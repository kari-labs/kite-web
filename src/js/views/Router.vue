<template>
    <v-app>
        <neit-header></neit-header>
        <v-content id="content" class="scroll-y">
            <router-view></router-view>
        </v-content>
        <neit-footer></neit-footer>
    </v-app>
</template>

<script>
    export default {
        components: {
            'neit-header': () => import(/* webpackChunkName: "initialLoad" */ '../components/Header.vue'),
            'neit-footer': () => import(/* webpackChunkName: "initialLoad" */ '../components/Footer.vue')
        }
    }
</script>