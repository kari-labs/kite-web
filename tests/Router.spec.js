import { createLocalVue, shallowMount } from '@vue/test-utils'
import VueRouter from 'vue-router'
import Vuetify from 'vuetify'
import Router from '../src/js/views/Router.vue'
import Header from '../src/js/components/Header.vue'
import Footer from '../src/js/components/Footer.vue'

describe('Router', () => {
    let testBed;

    const routes = [
        {
            path: '/',
            name: 'listContainers'
        }
    ]

    const router = new VueRouter({ routes })

    beforeEach(() => {
        const localVue = createLocalVue()
        localVue.config.productionTip = false
        localVue.use(VueRouter)
        localVue.use(Vuetify)

        testBed = shallowMount(Router, {
            stubs: {
                'neit-header': Header,
                'neit-footer': Footer
            },
            localVue,
            router
        })
    })

    it('should render site header', () => {
        expect(testBed.contains('nav.v-toolbar')).toBe(true)
    })

    it('should render site footer', () => {
        expect(testBed.contains('footer.v-footer')).toBe(true)
    })

    it('should render correctly', () => {
        expect(testBed.element).toMatchSnapshot()
    })
})