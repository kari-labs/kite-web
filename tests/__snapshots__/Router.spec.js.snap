// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Router should render correctly 1`] = `
<vapp-stub
  id="app"
>
  <div>
    <aside
      class="accent v-navigation-drawer v-navigation-drawer--clipped v-navigation-drawer--close v-navigation-drawer--fixed v-navigation-drawer--is-mobile theme--light"
      style="height: 100%; margin-top: 0px; max-height: calc(100% - 0px); transform: translateX(-300px); width: 300px;"
    >
      <div
        class="v-list pt-0 v-list--dense theme--light"
      >
        <div>
          <a
            class="v-list__tile v-list__tile--link theme--light"
          >
            <div
              class="v-list__tile__action"
            >
              <i
                aria-hidden="true"
                class="v-icon material-icons theme--light"
              >
                dashboard
              </i>
            </div>
             
            <div
              class="v-list__tile__content"
            >
              <div
                class="v-list__tile__title"
              >
                Home
              </div>
            </div>
          </a>
        </div>
        <div>
          <a
            class="v-list__tile v-list__tile--link theme--light"
          >
            <div
              class="v-list__tile__action"
            >
              <i
                aria-hidden="true"
                class="v-icon material-icons theme--light"
              >
                bug_report
              </i>
            </div>
             
            <div
              class="v-list__tile__content"
            >
              <div
                class="v-list__tile__title"
              >
                Report Bug
              </div>
            </div>
          </a>
        </div>
      </div>
      <div
        class="v-navigation-drawer__border"
      />
    </aside>
     
    <nav
      class="v-toolbar v-toolbar--clipped v-toolbar--fixed theme--dark primary"
      relative=""
      style="margin-top: 0px; padding-right: 0px; padding-left: 0px; transform: translateY(0px);"
    >
      <div
        class="v-toolbar__content"
        style="height: 64px;"
      >
        <button
          aria-label="Toggle side navigation drawer"
          class="secondary-text v-toolbar__side-icon v-btn v-btn--icon theme--dark"
          temporary=""
          type="button"
        >
          <div
            class="v-btn__content"
          >
            <i
              aria-hidden="true"
              class="v-icon material-icons theme--dark"
            >
              $vuetify.icons.menu
            </i>
          </div>
        </button>
         
        <div
          class="v-toolbar__title secondary-text"
        >
          NEIT Kite
        </div>
         
        <div
          class="spacer"
        />
         
        <div
          class="v-toolbar__items hidden-sm-and-down"
        >
          <button
            aria-label="Container Dashboard"
            class="v-btn v-btn--flat theme--dark secondary--text"
            slot="activator"
            type="button"
          >
            <div
              class="v-btn__content"
            >
              Container Dashboard
            </div>
          </button>
           
          <button
            aria-label="Report a Bug"
            class="v-btn v-btn--flat theme--dark secondary--text"
            type="button"
          >
            <div
              class="v-btn__content"
            >
              Report Bug
            </div>
          </button>
        </div>
      </div>
    </nav>
  </div>
   
  <vcontent-stub
    class="scroll-y"
    id="content"
    tag="main"
  >
    <router-view-stub
      name="default"
    />
  </vcontent-stub>
   
  <footer
    class="v-footer v-footer--absolute v-footer--inset theme--light primary"
    style="height: auto;"
  >
    <div
      class="text-xs-center v-card v-card--flat v-card--tile theme--light primary"
    >
      <div
        class="v-card__text"
      >
        <button
          aria-label="fab fa-facebook"
          class="mx-3 secondary-text v-btn v-btn--icon theme--light"
          type="button"
        >
          <div
            class="v-btn__content"
          >
            <i
              aria-hidden="true"
              class="v-icon fab fa-facebook theme--light secondary--text"
              style="font-size: 24px;"
            />
          </div>
        </button>
        <button
          aria-label="fab fa-twitter"
          class="mx-3 secondary-text v-btn v-btn--icon theme--light"
          type="button"
        >
          <div
            class="v-btn__content"
          >
            <i
              aria-hidden="true"
              class="v-icon fab fa-twitter theme--light secondary--text"
              style="font-size: 24px;"
            />
          </div>
        </button>
        <button
          aria-label="fab fa-google-plus"
          class="mx-3 secondary-text v-btn v-btn--icon theme--light"
          type="button"
        >
          <div
            class="v-btn__content"
          >
            <i
              aria-hidden="true"
              class="v-icon fab fa-google-plus theme--light secondary--text"
              style="font-size: 24px;"
            />
          </div>
        </button>
        <button
          aria-label="fab fa-instagram"
          class="mx-3 secondary-text v-btn v-btn--icon theme--light"
          type="button"
        >
          <div
            class="v-btn__content"
          >
            <i
              aria-hidden="true"
              class="v-icon fab fa-instagram theme--light secondary--text"
              style="font-size: 24px;"
            />
          </div>
        </button>
      </div>
       
      <div
        class="v-card__text pt-0 secondary-text"
      >
        
            Use of this system is limited to authorized persons of New England Institute of Technology.
        
      </div>
       
      <hr
        class="v-divider theme--light"
      />
       
      <div
        class="v-card__text secondary-text"
      >
        
            ©2018 — 
        <strong>
          Joseph Zurowski
        </strong>
         
        <br />
      </div>
    </div>
  </footer>
</vapp-stub>
`;
